src/server/pfs/server/driver.go

```diff
@@ -1747,25 +1692,48 @@ func (d *driver) getTreeForFile(ctx context.Context, file *pfs.File) (hashtree.H
 	if err != nil {
 		return nil, err
 	}
-	// Read everything under the scratch space for this commit
-	resp, err := d.etcdClient.Get(ctx, prefix, etcd.WithPrefix(), etcd.WithSort(etcd.SortByModRevision, etcd.SortAscend))
-	if err != nil {
-		return nil, err
-	}
-
 	parentTree, err := d.getTreeForCommit(ctx, commitInfo.ParentCommit)
 	if err != nil {
 		return nil, err
 	}
-	openTree := parentTree.Open()
-	if err := d.applyWrites(resp, openTree); err != nil {
-		return nil, err
-	}
-	tree, err := openTree.Finish()
+	return d.constructTreeFromPrefix(ctx, prefix, parentTree)
+}
+
+func (d *driver) constructTreeFromPrefix(ctx context.Context, prefix string, parentTree hashtree.HashTree) (hashtree.HashTree, error) {
```

Comment:
This is a bit pedantic but I find the name `constructTreeFromPrefix` a little incongruous with the other names in this file. For one thing I think you should use `get` in place of `construct` since they have about the same connotation and `get` is what other similar functions use. Another thing is that it doesn't quite feel like it's constructing the tree "from" a prefix. Maybe `getTreeForPrefix`, how would you feel about that?

Reply:
Sounds good :+1: